plugins/file/file_output.go

```diff
@@ -126,14 +151,28 @@ func (o *FileOutput) Init(config interface{}) (err error) {
 }
 
 func (o *FileOutput) openFile() (err error) {
-	basePath := filepath.Dir(o.Path)
+	basePath := filepath.Dir(o.path)
 	if err = os.MkdirAll(basePath, o.folderPerm); err != nil {
 		return fmt.Errorf("Can't create the basepath for the FileOutput plugin: %s", err.Error())
 	}
 	if err = plugins.CheckWritePermission(basePath); err != nil {
 		return
 	}
-	o.file, err = os.OpenFile(o.Path, os.O_WRONLY|os.O_APPEND|os.O_CREATE, o.perm)
+	o.file, err = os.OpenFile(o.path, os.O_WRONLY|os.O_APPEND|os.O_CREATE, o.perm)
+	return
+}
+
+func (o *FileOutput) tryRotateFile() (err error) {
+	if o.RotationInterval == 0 {
+		return
+	}
+	t := time.Now().Truncate(time.Duration(o.RotationInterval) * time.Hour)
```

Comment:
Generally, your code looks good for me (as an independent reviewer). But, i was thinking about different approach to the problem caused by performance concerns (however, its only my suggestion). Did you thought about using timer to create separate go function which would be called periodically (in periods defined by user) instead of calling it for every write?
Also it would be crucial to create some tests for this new feature.
However, thats just my thoughts @rafrombrc is more important. Anyway, good job and keep on hacking :)